### PR TITLE
Moved files to `src` and `resources`

### PR DESCRIPTION
**Pull request by gmarceau**
Getting ready to bring the callback implementation.

### CODE REVIEW

File: src/contract.impl.js [hunk 1]
```diff
@@ -0,0 +1,1209 @@
+// -*- js-indent-level: 2 -*-
```
Comment (paulmelnikow) [inline src/contract.impl.js]: Hmm, were these duplicated instead of renamed?

Comment (gmarceau) [inline src/contract.impl.js]: Ousp, 😛  fixed now

File: src/contract.impl.js [hunk 2]
```diff
@@ -1013,16 +1013,10 @@ function fnHelper(who, argumentContracts) {
...
-        _.each(prototypeFields, function (c, k) {
+        _.each(prototypeFields, function (v, k) {
           var freshContext = _.clone(context);
           freshContext.thingName = k;
-          if (c.thisContract === any) {
-            // Functions with no specified `thisContract` are assumed to be methods
-            // and given a `thisContract`
-            c = gentleUpdate(c, { thisContract: newThisContract });
-          }
-          WrappedConstructor.prototype[k] = checkWrapWContext(c, WrappedConstructor.prototype[k], freshContext);
```
Comment (paulmelnikow) [inline src/contract.impl.js]: Were these changes deliberate?

Comment (gmarceau) [inline src/contract.impl.js]: Good catch. Commit `ff6586` got dropped in the rebase.

**Comment by paulmelnikow** [comment]
Okay, looking good! 👍
